components/director/internal/open_resource_discovery/validation.go

```diff
@@ -192,43 +268,64 @@ func validateEventInput(event *model.EventDefinitionInput) error {
+			}),
+		),
+		validation.Field(&event.Industry,
+			validation.By(func(value interface{}) error {
+				return validateJSONArrayOfStringsContainsInMap(value, Industries)
+			}),
+			validation.By(func(value interface{}) error {
+				return validateJSONArrayOfStringsMatchPattern(value, regexp.MustCompile(StringArrayElementRegex))
+			}),
+		),
+		validation.Field(&event.ResourceDefinitions, validation.By(func(value interface{}) error {
+			return validateEventResourceDefinition(value, *event, packagePolicyLevels)
 		})),
-		validation.Field(&event.ResourceDefinitions, validation.Required),
 		validation.Field(&event.Links, validation.By(validateORDLinks)),
 		validation.Field(&event.ReleaseStatus, validation.Required, validation.In(ReleaseStatusBeta, ReleaseStatusActive, ReleaseStatusDeprecated)),
 		validation.Field(&event.SunsetDate, validation.When(*event.ReleaseStatus == ReleaseStatusDeprecated, validation.Required), validation.When(event.SunsetDate != nil, validation.By(isValidDate(event.SunsetDate)))),
-		validation.Field(&event.Successor, validation.When(*event.ReleaseStatus == ReleaseStatusDeprecated, validation.Required), validation.Match(regexp.MustCompile(EventOrdIDRegex))),
+		validation.Field(&event.Successors, validation.When(*event.ReleaseStatus == ReleaseStatusDeprecated, validation.Required), validation.By(func(value interface{}) error {
+			return validateJSONArrayOfStringsMatchPattern(value, regexp.MustCompile(EventOrdIDRegex))
+		})),
 		validation.Field(&event.ChangeLogEntries, validation.By(validateORDChangeLogEntries)),
 		validation.Field(&event.Labels, validation.By(validateORDLabels)),
 		validation.Field(&event.PartOfConsumptionBundles, validation.By(func(value interface{}) error {
 			return validateEventPartOfConsumptionBundles(value, regexp.MustCompile(BundleOrdIDRegex))
 		})),
-		validation.Field(&event.Extensible, validation.By(validateExtensibleField)),
+		validation.Field(&event.Extensible, validation.By(func(value interface{}) error {
+			return validateExtensibleField(value, event.OrdPackageID, packagePolicyLevels)
+		})),
 	)
 }
 
 func validateProductInput(product *model.ProductInput) error {
+	productOrdIDNamespace := strings.Split(product.OrdID, ":")[0]
+
 	return validation.ValidateStruct(product,
 		validation.Field(&product.OrdID, validation.Required, validation.Match(regexp.MustCompile(ProductOrdIDRegex))),
 		validation.Field(&product.Title, validation.Required),
 		validation.Field(&product.ShortDescription, shortDescriptionRules...),
-		validation.Field(&product.Vendor, validation.Required, validation.Match(regexp.MustCompile(VendorOrdIDRegex))),
+		validation.Field(&product.Vendor, validation.Required,
+			validation.Match(regexp.MustCompile(VendorOrdIDRegex)),
+			validation.When(regexp.MustCompile("^(sap)((\\.)([a-zA-Z0-9])+)*$").MatchString(productOrdIDNamespace), validation.In(SapVendor)).Else(validation.NotIn(SapVendor)),
```

Comment:
Shouldn't the part of the regex after the `.` be the same as the one defined in the spec: `[a-zA-Z0-9._\-]+`?